### ISSUE TITLE
RuleHandler.js is missing from the latest template in altinn studio

### ISSUE THREAD

**Issue by ivarne** [Altinn/app-frontend-react#870]
### Description of the bug

I think this should be fixed by changing app-frontend-react to not crash when `{org}/{app}/api/rulehandler/{id}` responds with 404.

The nuget packages in app-lib-dotnet should probably also return `204 No Content` instead of `404 Not Found` so that it doesn't trash the browser error logs.

### Steps To Reproduce

1. Create a new app in altinn studio
2. See that `RuleHandler.js` is missing from the template
3. Start the app
4. See that frontend crashes because {org}/{app}/api/rulehandler/{id} replies 404

**Comment by olemartinorg** [comment]
I agree. This should be an easy fix, and we're getting real close to the next release anyway. Adding this to the current sprint, and letting @bjosttveit and @Magnusrm know so they can chase me to it. :wink:

**Comment by ivarne** [comment]
I’ll make a PR for the status code change in backend. Missing this file is not an error anymore.

**Comment by olemartinorg** [comment]
Turns out it's not as critical as we first thought. The file [exists in the template](https://github.com/Altinn/app-template-dotnet/blob/main/src/App/ui/RuleHandler.js), but was gone when an app was copied from `altinn2convert`.

Link to slack discussion:
- https://altinn.slack.com/archives/C02EJ9HKQA3/p1674818338937379

**Comment by ivarne** [comment]
Seems to me like the only thing required here is to make the backend reply with `204 No Content` instead of `404 Not Found` when `RuleHandler.js` is missing. Same solution seems to be used with `RuleConfiguration.json` that currently isn't in the template. (though it returns `200 Ok` with an empty response). Not sure what status code makes most sense, but in https://github.com/Altinn/app-lib-dotnet/pull/174 I decided to go with `204 No Content` for both files.

**Comment by ivarne** [comment]
With https://github.com/Altinn/app-lib-dotnet/pull/174 merged (and soon released in the Altinn.App.Api nuget package), the only task left on this one is for @Altinn/team-altinnstudio to decide when to actually remove the file. Maybe this issue should be moved to altinn-studio?

**Comment by olemartinorg** [comment]
Right. I believe removing the file as a default is an issue that should be handled by the apps and studio teams in unison (it's an apps thing, but needs studio support before we remove it). Sadly, expressions still don't support calculations, and there is not a 100% overlap in supported functionality, so we have no pressing plans to remove these files from our template AFAIK.

Btw, the template application is located in this repo: https://github.com/Altinn/app-template-dotnet
